backend/Squirrel.Core/Squirrel.Core.BLL/Services/AzureStorageService.cs

```diff
@@ -0,0 +1,96 @@
+﻿using AutoMapper;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Squirrel.Core.BLL.Interfaces;
+using Squirrel.Core.Common.DTO.Blob;
+using Squirrel.Core.DAL.Context;
+
+namespace Squirrel.Core.BLL.Services
+{
+    public class AzureStorageService : BaseService, IStorageService
+    {
+        private readonly BlobServiceClient _blobServiceClient;
+
+        public AzureStorageService(SquirrelCoreContext context, IMapper mapper, BlobServiceClient blobServiceClient) : base(context, mapper)
+        {
+            _blobServiceClient = blobServiceClient;
+        }
+
+        public async Task DeleteAsync(string container, string id)
+        {
+            var containerClient = await GetOrCreateContainerByNameAsync(container);
+            var blobClient = containerClient.GetBlobClient(id);
+
+            if (!await blobClient.ExistsAsync())
+            {
+                throw new InvalidOperationException($"Blob with id:{id} does not exist.");
+            }
+            await blobClient.DeleteAsync();
+        }
+
+        public async Task<BlobDto> DownloadAsync(string container, string id)
+        {
+            var containerClient = await GetOrCreateContainerByNameAsync(container);
+            var blobClient = containerClient.GetBlobClient(id);
+
+            if(await blobClient.ExistsAsync())
+            {
+                var content = await blobClient.DownloadContentAsync();
```

Comment:
I think it's not necessary, because this method already uses the using statement inside of it. DownloadContentInternal: `using BlobDownloadStreamingResult blobDownloadStreamingResult = response.Value;`

Reply:
You are right, sorry, I confused the methods